### PR TITLE
feat: add LDAP signin method

### PR DESCRIPTION
**Pull request by HGZ-20**
Add support for LDAP login methods
Add option to control LDAP user in password login method.

Fix: https://github.com/casbin/casdoor/issues/2472

https://github.com/casbin/casdoor/assets/46803939/c82ab9fc-9650-42cf-abd1-843650b3d646

### CODE REVIEW

File: controllers/auth.go [hunk 1]
```diff
@@ -399,10 +399,14 @@ func (c *ApiController) Login() {
 				c.ResponseError(fmt.Sprintf(c.T("auth:The application: %s does not exist"), authForm.Application))
 				return
 			}
-			if !application.IsPasswordEnabled() {
+			if authForm.SigninMethod == "password" && !application.IsPasswordEnabled() {
```
Comment (hsluoyz) [inline controllers/auth.go:402]: Values as follows

W6{$Y55Q){%@ DVVZ){0XB

File: object/application.go [hunk 2]
```diff
@@ -544,6 +554,19 @@ func (application *Application) IsPasswordEnabled() bool {
 	}
 }
 
+func (application *Application) IsPasswordWithLdapEnabled() bool {
+	if len(application.SigninMethods) == 0 {
+		return application.EnablePassword
+	} else {
+		for _, signinMethod := range application.SigninMethods {
+			if signinMethod.Name == "Password" && signinMethod.Rule == "All" {
```
Comment (hsluoyz) [inline object/application.go:562]: "None" should also work like "All"

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: controllers/auth.go (hunk 1 above)
Comment (HGZ-20) [reply controllers/auth.go]: Done

File: object/application.go (hunk 2 above)
Comment (HGZ-20) [reply object/application.go:554]: Because the None value only affects new users within a week, compatibility with the None option will be dropped

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**